src/UI/Component/Layout/Page/Cockpit/Factory.php

```diff
@@ -0,0 +1,121 @@
+<?php
+namespace ILIAS\UI\Component\Layout\Page\Cockpit;
+/**
+ * This is the factory for cockpit-components
+ */
+interface Factory {
+    /**
+     * ---
+     * description:
+     *   purpose: >
+     *     The bar is the primary visual manifestation of the cockpit.
+     *
+     *   composition: >
+     *     The bar is the cockpit's contanier for entries.
```

Comment:
> yes, but this is not really a concern of UI?

It is to the extend to which the UI-Component and the according rules have to be flexible enough to handle all possible configuration. The configuration mechanics are not part of this though.